### PR TITLE
Migrate legacy dynamic attributes

### PR DESCRIPTION
**Pull request by BhavinPatel04**
## Description
Added a migrator to deprecate `<div ${attributes} />` in favor of using `<div ...attributes />`
Added test cases for the above

## Checklist:

- [x] I have read the **CONTRIBUTING** document and have signed (or will sign) the CLA.
- [ ] I have updated/added documentation affected by my changes.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: src/taglibs/migrate/all-tags/dynamic-attributes.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module.exports = function migrate(el, context) {
+    el.forEachAttribute(attr => {
+        const name = attr.name;
+        const value = attr.value;
+        if (
+            name == undefined &&
```
Comment (DylanPiercey) [inline src/taglibs/migrate/all-tags/dynamic-attributes.js]: This should use the same check that is used [here](https://github.com/marko-js/marko/blob/7d37fe36347fc75deb4148e9196f57e51b13121c/src/taglibs/core/core-transformer.js#L134) (`!attr.name && !attr.spread`). The implementation here will fail if you have something like `...({ a : 1 })` or any other type of node that isn't and identifier or member expression.

Also the previous warning should be removed from [here](https://github.com/marko-js/marko/blob/7d37fe36347fc75deb4148e9196f57e51b13121c/src/taglibs/core/core-transformer.js#L134) as well.

File: src/taglibs/migrate/all-tags/dynamic-attributes.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            (value.type == "MemberExpression" || value.type == "Identifier")
+        ) {
+            context.deprecate(
+                'The "${attributes}" is deprecated. Please use "...attributes" modifier instead. See: https://github.com/marko-js/marko/wiki/Deprecation:-w-*-Atrributes'
+            );
+            attr.spread = true;
```
Comment (DylanPiercey) [inline src/taglibs/migrate/all-tags/dynamic-attributes.js]: This line can be removed if we switch to removing the attribute and adding the spread attribute to the end.

File: src/taglibs/migrate/all-tags/dynamic-attributes.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            if (value.property && value.property.value == "*") {
```
Comment (DylanPiercey) [inline src/taglibs/migrate/all-tags/dynamic-attributes.js]: This check should not be needed, to should be safe to just remove the existing dynamic attribute and add a new spread attribute like so:

```js
el.removeAttribute(attrName);
el.addAttribute({ spread: true, value: value });
```

File: test/render/fixtures/attrs/expected.html [hunk 4]
```diff
@@ -1 +1 @@
-<div data-encoding="&quot;hello&quot;" style="background-color: #FF0000; &lt;test>" class="my-div" checked>Hello World!</div>
```
Comment (DylanPiercey) [inline test/render/fixtures/attrs/expected.html]: This test should not need to change, doing so would be a breaking change. If the order of the dynamic tags moves to the end this should be fixed.

File: test/render/fixtures/script-nonce/expected.html [hunk 5]
```diff
@@ -1,3 +1,3 @@
-<script foo="bar" nonce="foo" hello="world">
```
Comment (DylanPiercey) [inline test/render/fixtures/script-nonce/expected.html]: This test should not need to be changed.

**Comment by BhavinPatel04** [comment]
Closing this since a new invalid user `Patel` is added to committers.

**Pull request closed without merging**